### PR TITLE
Fix svsim DPI function template

### PR DESCRIPTION
**Pull request by uenoku**
This fixes two bugs in https://github.com/chipsalliance/chisel/pull/4286

* VCS seems to be unhappy with `initTestBenchScope;` so replace it with `initTestBenchScope();`. 
* `getBitWidth` exported DPI functions are used both for gettable/settable ports but #4286 only added DPI decl for getter so this PR fixes by removing a filter. 

### Contributor Checklist

- [x] Did you add Scaladoc to every public function/method?
- [ ] Did you add at least one test demonstrating the PR?
- [x] Did you delete any extraneous printlns/debugging code?
- [x] Did you specify the type of improvement?
- [ ] Did you add appropriate documentation in `docs/src`?
- [x] Did you request a desired merge strategy?
- [x] Did you add text to be included in the Release Notes for this change?

#### Type of Improvement

- Bugfix

#### Desired Merge Strategy

- Squash: The PR will be squashed and merged (choose this if you have no preference).

### Reviewer Checklist (only modified by reviewer)
- [ ] Did you add the appropriate labels? (Select the most appropriate one based on the "Type of Improvement")
- [ ] Did you mark the proper milestone (Bug fix: `3.6.x`, `5.x`, or `6.x` depending on impact, API modification or big change: `7.0`)?
- [ ] Did you review?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?
- [ ] Did you do one of the following when ready to merge:
  - [ ] Squash: You/ the contributor `Enable auto-merge (squash)`, clean up the commit message, and label with `Please Merge`.
  - [ ] Merge: Ensure that contributor has cleaned up their commit history, then merge with `Create a merge commit`.

### CODE REVIEW

**Review by mwachs5 — APPROVED** [review]

**Comment by mwachs5** [comment]
is there a testcase we can add for this ?

File: svsim/src/main/scala/Workspace.scala [hunk 1]
```diff
@@ -243,9 +243,11 @@ final class Workspace(
       l()
       l("extern \"C\" {")
       l(" svScope setScopeToTestBench();")
+      for ((port, index) <- ports) {
+      l(CreateFunctionForPort.createGetBitWidth(port.name))
+      }
       for ((port, index) <- ports.filter(_._1.isGettable)) {
       l(CreateFunctionForPort.createGetBits(port.name))
```
Comment (jackkoenig) [inline svsim/src/main/scala/Workspace.scala:250]:
```suggestion
      for ((port, index) <- ports) {
      l(CreateFunctionForPort.createGetBitWidth(port.name))
        if (port.isGettable) {
      l(CreateFunctionForPort.createGetBits(port.name))
        }
```
Stylistically, I think these would be better in a single loop (it also would keep the functions for a given port next to each other).

**Comment by uenoku** [comment]
> is there a testcase we can add for this ?

I tried but it looks tricky to reproduce the error in chisel repo. Internally we use svsim in a slightly different way and it looks there is no issue if we used `chisel3.simulator` directly.

File: svsim/src/main/scala/Workspace.scala (hunk 1 above)
Comment (uenoku) [reply svsim/src/main/scala/Workspace.scala]: Fused a setter and getter generation into a single loop.

**Review by jackkoenig — APPROVED** [review]

**Comment by sequencer** [comment]
> I tried but it looks tricky to reproduce the error in chisel repo. Internally we use svsim in a slightly different way and it looks there is no issue if we used chisel3.simulator directly.

Our team do provide the VCS infrastructure for rocket-chip and T1. Do we need to adopt it for chisel?

**Comment by uenoku** [comment]
> Our team do provide the VCS infrastructure for rocket-chip and T1. Do we need to adopt it for chisel?

No I don't think so. It was not backend specific and internally we use some wrapper for svsim that manually constructs `ModuleInfo` .

**Pull request merged**